jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/StageCancelTest.java

```diff
@@ -0,0 +1,214 @@
+                                   new CancelNodeInstanceCommand(processInstanceId, activeStageNode.next().getId()));
+            
+            //Get all stages
+            allStages = caseRuntimeDataService.getCaseInstanceStages(caseId, false, new QueryContext());
+            
+            assertThat(allStages)
+            .hasSize(4)
+            .anyMatch(item -> item.getStatus().equals(StageStatus.Available))
+            .anyMatch(item -> item.getStatus().equals(StageStatus.Canceled))
+            .noneMatch(item -> item.getStatus().equals(StageStatus.Completed))
+            .filteredOn(item -> item.getStatus().equals(StageStatus.Active)).hasSize(2);
+            
+            assertActiveStages(caseId, 2);
+            
+            //2. Complete Stage two
+            caseService.triggerAdHocFragment(caseId, "Stage two", "Simple task", null);
+            List<TaskSummary> tasks = runtimeDataService.getTasksAssignedAsPotentialOwner("john", new QueryFilter());
+            assertThat(tasks).isNotNull().hasSize(1);
+            assertTask(tasks.get(0), "john", "Simple task", Status.Reserved);
+
+            userTaskService.completeAutoProgress(tasks.get(0).getId(), "john", new HashMap<>());
+
+            //Get all stages
+            allStages = caseRuntimeDataService.getCaseInstanceStages(caseId, false, new QueryContext());
+            
+            assertThat(allStages)
+            .hasSize(4)
+            .anyMatch(item -> item.getStatus().equals(StageStatus.Available))
+            .anyMatch(item -> item.getStatus().equals(StageStatus.Active))
+            .anyMatch(item -> item.getStatus().equals(StageStatus.Canceled))
+            .anyMatch(item -> item.getStatus().equals(StageStatus.Completed));
+            
+            assertActiveStages(caseId, 1);
+            
+            //3. Cancel Stage three
+            activeStageNode.next();
+            processService.execute(caseInstance.getDeploymentId(), ProcessInstanceIdContext.get(processInstanceId), 
+                                   new CancelNodeInstanceCommand(processInstanceId, activeStageNode.next().getId()));
+            
+            //Get all stages
+            allStages = caseRuntimeDataService.getCaseInstanceStages(caseId, false, new QueryContext());
+            
+            assertThat(allStages)
+            .hasSize(4)
+            .anyMatch(item -> item.getStatus().equals(StageStatus.Available))
+            .anyMatch(item -> item.getStatus().equals(StageStatus.Completed))
+            .noneMatch(item -> item.getStatus().equals(StageStatus.Active))
+            .filteredOn(item -> item.getStatus().equals(StageStatus.Canceled)).hasSize(2);
+            
+            assertActiveStages(caseId, 0);
+
+        } catch (Exception e) {
+            fail("Unexpected exception " + e.getMessage());
+        } finally {
+            if (caseId != null) {
```

Comment:
this check is not needed, as caseId will never be equal to null at this point due to `assertNotNull(caseId);` at the beginning